### ISSUE TITLE
i3blocks stops printing output after clicking on i3bar

### ISSUE THREAD

**Issue by bookletchoir** [vivien/i3blocks#193]
When I clicked on the status bar created by i3blocks on i3bar, the status bar froze and stopped updating values, while the workspaces on the left was still fine.

From the i3bar's log, I can tell it didn't received any futher values from i3blocks.

Here's my i3blocks config [[link]](https://gist.githubusercontent.com/bookletchoir/34d8c12d5b6be4c4919488bd4fc1c620/raw/63a770e9557c9d16b527bea3f62c636f51c9683a/i3blocks.conf), the script I used [[link]](https://gist.githubusercontent.com/bookletchoir/34d8c12d5b6be4c4919488bd4fc1c620/raw/63a770e9557c9d16b527bea3f62c636f51c9683a/i3blocksgen), and the i3bar log file [[link]](https://gist.githubusercontent.com/bookletchoir/f969a48b74af028e77a93669a8cfdd69/raw/3075e252db8e399cf85c3b2db09f206dd1f689a5/i3bar%20log).

For the script, it will receive the first arg, print the corresponding function and ignore the rest of the args and environment variables; and I can't find any problems with it.

For the log, I reproduced by running i3bar with i3blocks as status_command, wait for a few seconds and click on the status bar. It stopped updating values right after that. Then i switched workspaces (i3bar still response and redraw the workspaces), trigger the signal on the [volume] block, then stop the logging.

You can see at line 1128 of the log, i3bar stopped receiving any futher values.

I'm using i3 version 4.12, and i3blocks version 1.4, on Arch linux.

Well... where did i get it wrong?

**Comment by vivien** [comment]
I tried your setup with the following command:

```
status_command PATH=$PATH:/tmp/bookletchoir i3blocks -vvv -c /tmp/bookletchoir/i3blocks.conf 2>/tmp/bookletchoir/i3blocks.err
```

(except for for sensors and volume because I don't have nvidia-smi and mpc) and it seems to work fine for me. Clicking on blocks update them (tested by adding %S to the date format).

Can you try use i3blocks with `-vvv 2>/tmp/i3blocks.err`, run `tail -f /tmp/i3blocks.err` in a terminal and restart i3? So that you can check if you see anything suspicious.

**Comment by bookletchoir** [comment]
Here's the log file, I tried to click on every blocks: [[link]](https://gist.github.com/anonymous/56c24b3f6dda85c47847616f30474a20#file-i3blocks-err-L994)

It starts from line 994.

**Comment by vivien** [comment]
Hum, except the empty first click:
```
DEBUG click_parse:65: parsed click: name= instance= button= x= y=
```
this looks ok...

**Comment by bookletchoir** [comment]
Well... I tried latest commit but still, no lucks.

The log _looks_ normal but as soon as I clicked on the status bar, i3block stopped updating values. (still printing debug event for mouseclick).

Workspace status and tray icons work fine, as far as I can see.

**Comment by bookletchoir** [comment]
Temporary solved by now, so I'll close this.

Here's my (updated) script and config: [[link]](https://gist.github.com/bookletchoir/5777edc6075020ac9106164747c84c0d)

Some explainations, if you're interested:
- First, I used `hddtemp -dF /dev/sda` as systemd service, and it will use port `7634` to print status.
- Then, from the script, I used system call `telnet 127.0.0.1 7634`, which will read hddtemp's status (line 
207-213)
- And last, I printed it along with cpu_temp and gpu_temp (line 313)

To reproduce:
- Run `hddtemp -dF /dev/sda` as root or system service.
- Change line 313 to `return "{0} {1} {2}".format(cpu_temp(), gpu_temp(False), hdd_temp(False))`.
- Restart i3wm status bar and click on blocklets, it'll immediately stopped printing values, while still printing mouse events.

I don't know which exactly is the case, but when running the script normally, I don't have any problems even when using wrong port for telnet, so I assume (maybe) it's i3blocks problem.

And... well, that's all I can find.
